### PR TITLE
Code

### PR DESCRIPTION
**Pull request by MarcoAurelioAntonio**
# RUBY
## In this branch we have done the followings:
- Added a Function to decode the morse code.
- Added a function to represent the alphabet.
- Put the algorithm to change the morse code to alphabetical words.

### DONE 💯 🎉

### CODE REVIEW

File: code.rb [hunk 1]
```diff
@@ -39,8 +39,7 @@
 }
 
 def morse_decoder(string)
```
Comment (Ismailco) [inline code.rb:41]: - [ ] You should create 3 methods `decode_char()` to decode a Morse code character, `decode_word()` that reuses decode_char() to decode an entire word in Morse code, and finally `decode()` method that reuses decode_word() to decode the entire message in Morse code. kindly make sure to implement the 3 methods to decode the morse code string

Screenshot_20221215_214846

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @MarcoAurelioAntonio, @Mhamad-Raad, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: code.rb [hunk 2]
```diff
@@ -1,46 +1,51 @@
...
+  '-.--' => 'y',
+  '--..' => 'z'
 }
 
-def morse_decoder(string)
-    string.split(' ').map { |x| @morse_code.key(x) }.join
+def decode(string)
+  coded_words = string.split('   ')
+  coded_letters = []
+  letters = []
```
Comment (tresorsawasawa) [inline code.rb]: - [ ] As mentioned by the previous reviewer and for the project requirements, kindly make sure you create 3 separate methods where:
  - The 1st decodes a Morse code character, takes a string parameter, and returns the corresponding character in uppercase (e.g.` decode_char(".-")` returns "A").
  - The 2nd decodes an entire word in Morse code, takes a string parameter, and returns the string representation. Every character in a word will be separated by a single space (e.g. `decode_word("-- -.--")` returns "MY").
  - The last one decodes the entire message in Morse code, takes a string parameter, and returns the string representation. Every word will be separated by 3 spaces (e.g. `decode("-- -.--   -. .- -- .")` returns "MY NAME").

I can suggest you structure your code like in the below example:
```ruby
morse_letters = {
    '.-' => 'a',
    '-...' => 'b',
    '-.-.' => 'c',
    '-..' => 'd',
    '.' => 'e',
    ...
 }

  def decode_char 
    # Code for the decode_char method here
  end

  def decode_word 
    # Code for the decode_word method here
  end

  def decode
    # Code for the decode method here
  end
```

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUEST CHANGES :recycle::recycle: 

Hi @MarcoAurelioAntonio  and @Mhamad-Raad 
Great work on making the changes requested by the previous reviewer  👏👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by muneebulrehman — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR

## Optional suggestions

- Consider updating your README file. It still contains screenshot of your old project.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**